server/src/test/java/org/opensearch/extensions/ExtensionsOrchestratorTests.java

```diff
@@ -273,13 +294,68 @@ public void testExtensionsInitialize() throws Exception {
             );
 
             extensionsOrchestrator.extensionsInitialize();
-
-            transportService.close();
-            ThreadPool.terminate(threadPool, 30, TimeUnit.SECONDS);
             mockLogAppender.assertAllExpectationsMatched();
         }
     }
 
+    public void testHandleExtensionRequest() throws Exception {
+
+        Path extensionDir = createTempDir();
+
+        final ClusterService clusterService = mock(ClusterService.class);
+        ClusterName clusterName = ClusterName.CLUSTER_NAME_SETTING.get(settings);
+
+        ExtensionsOrchestrator extensionsOrchestrator = new ExtensionsOrchestrator(settings, extensionDir);
+
+        transportService.registerRequestHandler(
+            ExtensionsOrchestrator.REQUEST_EXTENSION_CLUSTER_STATE,
+            ThreadPool.Names.GENERIC,
+            false,
+            false,
+            ExtensionRequest::new,
+            (request, channel, task) -> {
+                channel.sendResponse(new ClusterStateResponse(clusterName, null, false));
+                assertNotNull(extensionsOrchestrator.handleExtensionRequest(request));
+            }
+        );
+
+        transportService.registerRequestHandler(
+            ExtensionsOrchestrator.REQUEST_EXTENSION_CLUSTER_SETTINGS,
+            ThreadPool.Names.GENERIC,
+            false,
+            false,
+            ExtensionRequest::new,
+            (request, channel, task) -> {
+                channel.sendResponse(new ClusterSettingsResponse(clusterService));
+                assertNotNull(extensionsOrchestrator.handleExtensionRequest(request));
+            }
+        );
+
+        transportService.registerRequestHandler(
+            ExtensionsOrchestrator.REQUEST_EXTENSION_LOCAL_NODE,
+            ThreadPool.Names.GENERIC,
+            false,
+            false,
+            ExtensionRequest::new,
+            (request, channel, task) -> {
+                channel.sendResponse(new LocalNodeResponse(clusterService));
+                assertNotNull(extensionsOrchestrator.handleExtensionRequest(request));
+            }
+        );
+
+    }
+
+    public void testHandlerResponse() throws Exception {
+        ClusterName clusterName = new ClusterName("cluster-1");
+        ClusterStateResponse clusterStateResponse = new ClusterStateResponse(clusterName, null, false);
+        assertEquals(clusterStateResponse.getClusterName(), clusterName);
```

Comment:
This is testing the `ClusterStateResponse` which is good, but its not testing the handler response. 

Lets write unit tests for:
- `handleExtensionRequest` pass in a request and verify the response params it sends back.
- Test all responses, eg `ClusterStateResponse` in `ClusterStateResponseTest` file. 

Unit tests are usually for that specific class. Lets write new files for every different class we'd want to test.

Reply:
- Done. Mocked the class type to verify the response is correct
- We already have [ClusterStateResponseTest](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/test/java/org/opensearch/action/admin/cluster/state/ClusterStateResponseTests.java) file present in OpenSearch. Rest of the responses are handled when verifying the class type of return object of class type in handleExtensionRequest. 
- Can you please elaborate more on this?